.github/workflows/golang.yaml

```diff
@@ -30,54 +30,74 @@ jobs:
     steps:
     - uses: actions/checkout@v4
       name: Checkout code
+
     - name: Get Golang version
       id: vars
       run: |
         GOLANG_VERSION=$(./hack/golang-version.sh)
         echo "GOLANG_VERSION=${GOLANG_VERSION##GOLANG_VERSION := }" >> $GITHUB_ENV
+
     - name: Install Go
       uses: actions/setup-go@v5
       with:
         go-version: ${{ env.GOLANG_VERSION }}
+
     - name: Lint
       uses: golangci/golangci-lint-action@v6
       with:
         version: latest
         args: -v --timeout 5m
         skip-cache: true
+
     - name: Check golang modules
       run: | 
         make check-vendor
         make -C deployments/devel check-modules
+
   test:
     name: Unit test
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+
       - name: Get Golang version
         id: vars
         run: |
           GOLANG_VERSION=$(./hack/golang-version.sh)
           echo "GOLANG_VERSION=${GOLANG_VERSION##GOLANG_VERSION := }" >> $GITHUB_ENV
+
       - name: Install Go
         uses: actions/setup-go@v5
         with:
           go-version: ${{ env.GOLANG_VERSION }}
+
       - run: make test
```

Comment:
Should we change this to `make coverage` instead?

Reply:
Don't we *need* the `coverage.out` artifact?

Reply:
> Well, it actually makes sense from the perspective of a Makefile file functionality; make coverage would actually generate an output artifact. So I am not against it, I'll leave it as your call

Sorry, I don't understand what you're saying here. We have two targets:
* `make test`
* `make coverage`
* 
The first generates `coverage.out` and the second filters out coverage related to mocks to generate a `.no-mocks` file.

Reply:
Sorry I was not aware of the `make coverage` but for `codecov` we need the output file from `make test`